src/thor/trace_route_action.cc

```diff
@@ -465,20 +467,27 @@ thor_worker_t::map_match(Api& request) {
             // we skip input location points that are:
             // 1. not on the match results edge
             // 2. neither break points nor the disjoint (upgraded break) points
-            if (path_edge.edgeid != leg_destination_iter->edgeid ||
-                (options.shape(leg_destination_iter - match_results.begin()).type() !=
-                     valhalla::Location::kBreak &&
-                 options.shape(leg_destination_iter - match_results.begin()).type() !=
-                     valhalla::Location::kBreakThrough &&
-                 leg_destination_iter != disjoint_point)) {
+            if (path_edge.edgeid != leg_destination_iter->edgeid) {
+              continue;
+            } else if (options.shape(leg_destination_iter - match_results.begin()).type() !=
```

Comment:
instead of having if followed by else if, can we make these 2 separate ifs and put appropriate comments above each about what cases they are handling